pkgs/development/tools/devpod/default.nix

```diff
@@ -0,0 +1,185 @@
+{ lib
+, buildGoModule
+, copyDesktopItems
+, darwin
+, desktopToDarwinBundle
+, fetchFromGitHub
+, fetchYarnDeps
+, gtk3
+, installShellFiles
+, jq
+, libayatana-appindicator
+, libsoup
+, makeDesktopItem
+, mkYarnPackage
+, openssl
+, pkg-config
+, rust
+, rustPlatform
+, stdenv
+, testers
+, webkitgtk
+}:
+
+let
+  pname = "devpod";
+  version = "0.3.7";
+
+  src = fetchFromGitHub {
+    owner = "loft-sh";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-s5O7pcrynfwgye6NdcF6XOCRDf1VS9i8Z3ghOQnjiKg=";
+  };
+
+  meta = with lib; {
+    description = "Codespaces but open-source, client-only and unopinionated: Works with any IDE and lets you use any cloud, kubernetes or just localhost docker. ";
+    homepage = "https://devpod.sh";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ maxbrunet ];
+  };
+in
+rec {
+  cli = buildGoModule {
+    inherit version src pname meta;
+
+    vendorSha256 = null;
+
+    CGO_ENABLED = 0;
+
+    ldflags = [
+      "-X github.com/loft-sh/devpod/pkg/version.version=v${version}"
+    ];
+
+    excludedPackages = [ "./e2e" ];
+
+    nativeBuildInputs = [ installShellFiles ];
+
+    postInstall = ''
+      $out/bin/devpod completion bash >devpod.bash
+      $out/bin/devpod completion fish >devpod.fish
+      $out/bin/devpod completion zsh >devpod.zsh
+      installShellCompletion devpod.{bash,fish,zsh}
+    '';
+
+    passthru.tests.version = testers.testVersion {
+      package = pname;
+      version = "v${version}";
+    };
+  };
+
+  desktop =
+    let
+      frontend-build = mkYarnPackage {
+        inherit version;
+        pname = "devpod-frontend";
+
+        src = "${src}/desktop";
+
+        offlineCache = fetchYarnDeps {
+          yarnLock = "${src}/desktop/yarn.lock";
+          sha256 = {
+            stdenv-linux = "sha256-O/w7g+kNgWrzP7EZ4Dgzea81c8hsqR9MQ8Fgmp7IwlE=";
+            stdenv-darwin = "sha256-z1dPaS5g6xlzKEJzPkpeShB+dTJHgQbzjQ9Rq3o+Xjo=";
+          }."${stdenv.name}";
+        };
+
+        buildPhase = ''
+          export HOME=$(mktemp -d)
+          yarn --offline run build
+
+          cp -r deps/devpod/dist $out
+        '';
+
+        distPhase = "true";
+        dontInstall = true;
+      };
+
+      rustTargetPlatformSpec = rust.toRustTargetSpec stdenv.hostPlatform;
+    in
+    rustPlatform.buildRustPackage {
+      inherit version src;
+      pname = "devpod-desktop";
+
+      sourceRoot = "${src.name}/desktop/src-tauri";
+
+      cargoLock = {
+        lockFile = "${src}/desktop/src-tauri/Cargo.lock";
+        outputHashes = {
+          "tauri-plugin-log-0.1.0" = "sha256-Ei0j7UNzsK45c8fEV8Yw3pyf4oSG5EYgLB4BRfafq6A=";
+        };
+      };
+
+      # Workaround:
+      #   The `tauri` dependency features on the `Cargo.toml` file does not match the allowlist defined under `tauri.conf.json`.
+      #   Please run `tauri dev` or `tauri build` or add the `updater` feature.
+      patches = [ ./add-tauri-updater-feature.patch ];
```

Comment:
Is this issue tracked upstream? Please link to it.

Reply:
Not really, upstream is not interested in fixing it: loft-sh/devpod#648

I added a comment about it.

As Tauri gets more popular, it might become worth adding proper support for it in nixpkgs, it could just be about adding flexibility around what build command is used in `buildRustPackage`